### ISSUE TITLE
Lycon issues

### ISSUE THREAD

**Issue by k4kuz0** [eriklindernoren/PyTorch-YOLOv3#142]
Been having problems all day with Lycon, it doesn't appear to work on my installation of MacOS (Mojave 10.14.4). I've tried two approaches, both leading to errors:

I installed the dependencies shown here https://github.com/ethereon/lycon using homebrew, and tried using MacOS inbuilt python3, but whenever `lycon.load(file_path)` is run, it returns `None`, causing an error in `detect.py`.

Alternatively I tried installing everything with Anaconda, and I get an (even worse) segmentation fault when trying to import Lycon. 

Any idea what's going on?

Also, as mentioned in #102 there are several missing requirements in the requirements.txt as far as I can see, namely:

- tqdm
- scitech-image
- torch version is wrong (should be 1.0 as mentioned in #102)

**Comment by eriklindernoren** [comment]
I'll see if I can find a better module.  I wanted something that gives equal performance to OpenCV but without all the hassle of the OpenCV installation. Seems like multiple people have encountered issues installing Lycon though so I might have to go in that direction anyways. The speed-up over something like Pillow or scikit-image is significant so I don't like the idea using any of those for the dataloader. I'll make sure to update requirements.txt, thanks!

**Comment by eriklindernoren** [comment]
@k4kuz0 I just installed lycon successfully on my Mac (Mojave 10.14), by:
`$ brew install cmake`
`$ pip3 install lycon`

**Comment by k4kuz0** [comment]
I had no issue installing Lycon, but if I tried loading any image in it would only return `None`, is this the case for you on your Mac? I installed Lycon in pretty much the same way using homebrew and pip3.

**Comment by eriklindernoren** [comment]
@k4kuz0 No it loads images just fine for me. 

```
Eriks-MacBook-Pro-2:~ erik$ cd git/PyTorch-YOLOv3/
Eriks-MacBook-Pro-2:PyTorch-YOLOv3 erik$ python3
Python 3.7.3 (default, Apr  2 2019, 15:16:43) 
[Clang 10.0.0 (clang-1000.11.45.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import lycon
>>> lycon.load("data/samples/dog.jpg").shape
(576, 768, 3)
>>> 
```

**Comment by dirtshell** [comment]
I have a similar issue on Ubuntu 18.04 with lycon installed with pip. 

I get a crash at `util/datasets.py:89`. It states that `img` is `None`. But if I load up a Python shell and import lycon and then attempt to load the same image that it crashed on, it loads up just fine and I get the correct dimensions.

**Comment by dirtshell** [comment]
After digging around in this issue a bit more, it really seems like a memory issue, where `img` is getting overwritten somehow, and becomes `None`. I can wrap the whole thing in a repeating loop with the entire function in a `try` clause, but that just means that you hit an issue with other functions that reference `img`. Something to do with the way numpy has these act as forks or something?

**Comment by eriklindernoren** [comment]
That is strange. It's also hard for me to debug this as I can't reproduce this issue. Does the same thing happen if you use `PIL.Image`, @dirtshell ?

**Comment by k4kuz0** [comment]
@eriklindernoren `PIL.Image` works fine for me in loading the image, but Lycon produces `None`. I'm at a bit of a loss as to what I could have on my Mac that makes your installation work, but mine not. I tried downgrading my Python version to 3.5 as I noticed on the Lycon github page that they have only tested in macOS with Python 3.5, but to no avail.

**Comment by heatherlock** [comment]
I was experiencing the same error as @dirtshell When I pulled the latest from the repo, the error disappeared and everything ran smoothly.

**Comment by dirtshell** [comment]
IDK what changed, but I did a pull like @heatherlock recommended and I am no longer having the issue!

**Comment by eriklindernoren** [comment]
I'm not sure what recent change fixed this either to be honest. @k4kuz0 Did this work for you as well?

**Comment by eriklindernoren** [comment]
I decided to revert to `PIL.Image.open` for loading images and `torch.nn.functional.interpolate` for image resizing. I benchmarked the two approaches and they were close enough to motivate the change.
